task/clair-scan/0.1/clair-scan.yaml

```diff
@@ -24,7 +24,38 @@ spec:
       description: Tekton task test output.
     - name: CLAIR_SCAN_RESULT
       description: Clair scan result.
+    - name: IMAGES_PROCESSED
+      description: Images processed in the task.
   steps:
+    - name: get-image-manifests
+      image: quay.io/redhat-appstudio/hacbs-test:v1.3.0@sha256:cd4601a7d71ebd908046db7a9b7010611b8b372fe941664d5163c81250a1a1fc
+      env:
+        - name: IMAGE_URL
+          value: $(params.image-url)
+        - name: IMAGE_DIGEST
+          value: $(params.image-digest)
+      securityContext:
+        capabilities:
+          add:
+            - SETFCAP
+      script: |
+        #!/usr/bin/env bash
+
+        imagewithouttag=$(echo $IMAGE_URL | sed "s/\(.*\):.*/\1/" | tr -d '\n')
+        # strip new-line escape symbol from parameter and save it to variable
+        imageanddigest=$(echo $imagewithouttag@$IMAGE_DIGEST)
+        echo "Inspecting raw image manifest $imageanddigest."
+
+        # check if the image is a manifest list
+        skopeo_inspect_raw=$(skopeo inspect --no-tags --raw docker://$imageanddigest)
+        if echo "$skopeo_inspect_raw" | jq .mediaType | grep -q "oci.image.index"; then
+          for arch in $(echo "$skopeo_inspect_raw" | jq -r '.manifests[].platform.architecture'); do
```

Comment:
I did something similar just by using JQ

https://github.com/redhat-appstudio/build-definitions/blob/acacbe1a5b84031157069a4df91b55029543f0d6/task/inspect-image/0.1/inspect-image.yaml#L98

given that multiple tasks will need the same code for detecting manifest images, I'd like to have a reusable function in hacbs-test to be used in all tasks, instead writing ad-hoc code per task

Reply:
Added a function to do `skopeo inspect` on the images. Please let me know what you think: https://github.com/redhat-appstudio/hacbs-test/pull/227 

The reason I did not add function to filter the `sha` and` arch` is because each function might parse the output differently.

Reply:
I meant function to extract image manifests from index image, not replacement of skopeo inspect call